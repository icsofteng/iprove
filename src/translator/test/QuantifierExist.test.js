const {translate} = require('../z3')
const {translate: translate_latex} = require('../../translator/latex')

const test_constants = [{ value: 'P', varType: 'Bool' }, { value: 'Q', varType: 'Bool' }]
const test_rules = [
  {
    type: 'existential_quantifier',
    symbol: 'exist',
    variables: [{ type: "variable", value: "x" , varType: "Type"}],
    value: {
      type: 'binary',
      symbol: 'implies',
      lhs: {
        type: 'literal',
        value: 'P'
      },
      rhs: {
        type: 'literal',
        value: 'Q'
      }
    }
  },
  {
    type: 'literal',
    value: 'P'
  }
]

test('Quantifier Exist test', () => {
  expect(translate(test_rules, test_constants, [], [], [])).toMatchSnapshot()
})

test('Qualifier Exist test latex', () => {
  expect(translate_latex(test_rules, test_constants, [], [], [])).toMatchSnapshot()
})
