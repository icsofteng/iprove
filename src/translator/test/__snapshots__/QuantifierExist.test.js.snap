// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Qualifier Exist test latex 1`] = `
Array [
  "\\\\exist x:Type.P \\\\Longrightarrow Q",
  "P",
]
`;

exports[`Quantifier Exist test 1`] = `
"(declare-const P Bool)
(declare-const Q Bool)
(assert (exist ((x Type))(=> P Q)))
(assert (not P))
(check-sat)"
`;
